examples/data_access.py

```diff
@@ -1,13 +1,24 @@
-import pathlib
-from scipy import stats
-from data_pipeline_api.api import API, DataAccess, ParameterRead
-from data_pipeline_api.file_system_data_access import FileSystemDataAccess
+import pandas as pd
+from pathlib import Path
+from data_pipeline_api.simple_network_sim_api import SimpleNetworkSimAPI
 
-data_path = "../simple_network_sim/data_pipeline_inputs"
-api = API(FileSystemDataAccess(data_path, "metadata.toml"))
-population = api.read_table("human/population")
+"""
+TODO
 
-print(f"Population is {population['Total'].sum()}")
+[X] Make things work just off the config file?
+[X] Consider changing the config file to yaml.
+[X] Implement version handling.
+[ ] Add logging describing the process, to make it easier to debug failures.
+[ ] Work out the set of metadata keys.
+[ ] Flesh out the standard interface.
+[ ] Add up-front verification, to make sure we are not about to start a run that we
+    cannot complete.
+[ ] We shouldn't fail on overwriting a file, but rather a component?
+"""
 
-api.write_table(population, "output/table", version=1)
-api.close()
+with SimpleNetworkSimAPI(
+    "repos/data_pipeline_api/examples/test_data_2/config.yaml",
+) as api:
```

Comment:
Nice! Definitely think a context manager is more ergonomic over `api.close()`.

For the C++ bindings, we should be able to use RAII to achieve the same thing